android/src/main/java/io/ably/lib/util/AndroidPlatform.java

```diff
@@ -0,0 +1,71 @@
+package io.ably.lib.util;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.Build;
+
+import java.util.Properties;
+
+import io.ably.lib.util.Log;
+import io.ably.lib.util.Platform.PlatformEvent;
+
+public class AndroidPlatform extends Platform {
+	public static final String ANDROID_SDK_VERSION = "android-sdk-version";
+
+	public AndroidPlatform() {
+		name = "android";
+		properties.setProperty(ANDROID_SDK_VERSION, String.valueOf(Build.VERSION.SDK_INT));
+
+		try {
+			setContext((Context) Class.forName("android.app.ActivityThread").getMethod("currentApplication").invoke(null, (Object[]) null));
```

Comment:
Yes, I know. I think probably there is no way to do this purely as a library - you need to get the app to add something to its manifest (eg Service belonging to the library) or the application needs to call something in the library to pass in a `Context`.